x-pack/filebeat/module/cisco/amp/ingest/pipeline.yml

```diff
@@ -300,21 +309,41 @@ processors:
     value: "{{ cisco.amp.computer.external_ip }}"
     if: ctx?.cisco?.amp?.computer?.external_ip != null
     allow_duplicates: false
-- foreach:
-    field: cisco.amp.computer.network_addresses
-    processor:
-      append:
-        field: related.ip
-        value: "{{ _ingest._value.ip }}"
-        allow_duplicates: false
+- script:
+    lang: painless
+    source: |
+        if (ctx?.related == null) {
+            ctx.related = new HashMap();
+            ctx.related.ip = new ArrayList();
```

Comment:
u shouldn't need this line.  Its covered by the next if statement.

Reply:
Yeah those are correct :) Will add in the change.